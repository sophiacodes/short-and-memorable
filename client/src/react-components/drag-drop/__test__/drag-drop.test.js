import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import DragDrop from '../';

Enzyme.configure({ adapter: new Adapter() });

it('<DragDrop /> component to render correctly when loading', () => {
    const props = {
        loading: false,
        className: "drag-drop-container",
        onDrop: jest.fn(),
        onDragOver: jest.fn(),
        onDragEnter: jest.fn()
    };
    const tree = renderer
        .create(
            <DragDrop {...props} />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('<DragDrop /> component to render correctly when not loading', () => {
    const props = {
        loading: true,
        className: "drag-drop-container",
        onDrop: jest.fn(),
        onDragOver: jest.fn(),
        onDragEnter: jest.fn()
    };
    const tree = renderer
        .create(
            <DragDrop {...props} />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('<DragDrop /> component ensure onDrop.fn() can call prop', () => {
    const props = {
        loading: true,
        handleDrop: jest.fn()
    };
    const wrapper = mount(
      <DragDrop
        {...props}
      />);
    const e = { 
        stopPropagation: jest.fn(),
        preventDefault: jest.fn(),
        dataTransfer: {}
    };
    wrapper.find(DragDrop).instance().onDrop(e);

    expect(e.stopPropagation).toBeCalled();
    expect(e.preventDefault).toBeCalled();
    expect(wrapper.find('DragDrop').props().handleDrop).toEqual(props.handleDrop);
    expect(props.handleDrop).toHaveBeenCalledTimes(1);
});

it('<DragDrop /> component ensure onDragOver.fn() prevents default action', () => {
    const props = {
        loading: true,
        handleDrop: jest.fn()
    };
    const wrapper = mount(
      <DragDrop
        {...props}
      />);
    const e = { 
        stopPropagation: jest.fn(),
        preventDefault: jest.fn()
    };
    wrapper.find(DragDrop).instance().onDragOver(e);

    expect(e.stopPropagation).toBeCalled();
    expect(e.preventDefault).toBeCalled();
});

it('<DragDrop /> component ensure onDragEnter.fn() prevents default action', () => {
    const props = {
        loading: true,
        handleDrop: jest.fn()
    };
    const wrapper = mount(
      <DragDrop
        {...props}
      />);
    const e = { 
        stopPropagation: jest.fn(),
        preventDefault: jest.fn()
    };
    wrapper.find(DragDrop).instance().onDragEnter(e);

    expect(e.stopPropagation).toBeCalled();
    expect(e.preventDefault).toBeCalled();
});