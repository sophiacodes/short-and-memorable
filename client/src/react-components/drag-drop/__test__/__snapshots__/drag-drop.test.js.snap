// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DragDrop /> component to render correctly when loading 1`] = `
<div
  className="drag-drop-container "
  onDragEnter={[Function]}
  onDragOver={[Function]}
  onDrop={[Function]}
>
  <p
    className="drag-label"
  >
    Drag file to upload
  </p>
</div>
`;

exports[`<DragDrop /> component to render correctly when not loading 1`] = `
<div
  className="drag-drop-container loading"
  onDragEnter={[Function]}
  onDragOver={[Function]}
  onDrop={[Function]}
>
  <p
    className="drag-label"
  >
    Loading...
  </p>
</div>
`;
